### ISSUE TITLE
[doc] RFC 7946 GeoJSON

### ISSUE THREAD

**Issue by learning4life** [chris48s/arcgis2geojson#2]
Is the [RFC 7946 GeoJSON](https://sgillies.net/2016/08/11/rfc-7946-the-geojson-format.html) standard supported?

**Comment by chris48s** [comment]
Good question.

The short answer is 'no'. To give a very obvious example:

```python
>>> from arcgis2geojson import arcgis2geojson
>>> input = {
...     'x': -66.0123456789,
...     'y': 20.0123456789,
...     'spatialReference': {
...         'wkid': 4326
...     }
... }
>>> arcgis2geojson(input)
{'type': 'Point', 'coordinates': [-66.0123456789, 20.0123456789]}
```

In this case, I'm passing in an ArcGIS point object using >7 decimal places of precision and replicating them in the output GeoJSON. This violates the rule

> Don't write longitude and latitude values with more than 7 decimal places of precision.

There are probably also other violations of the spec you could point to, but that one is pretty easy to pick out and in itself makes the output non-compliant.

I do have some other thoughts, but I'm interested in your response at this point..

**Comment by learning4life** [comment]
True, arcgis2geojson dose follow the 2008 GeoJSON specifications and not RFC 7946 GeoJSON.

The [GDAL OGR GeoJSON](http://www.gdal.org/drv_geojson.html) driver supports the RFC 7946 using the RFC7946=YES option. But not even the new [ArcGIS Pro 2.1.1 supports the RFC 7946 GeoJSON specifications](https://community.esri.com/ideas/13080-arcgis-pro-direct-geojson-support), making it hard to locate an up-to-date ArcGIS GeoJSON solution. 

I like your idea, converting the ArcGIS JSON to GeoJSON could work with the RFC 7946 specifications as well.

**Comment by chris48s** [comment]
Spoiler alert: If you're looking for a converter that will go straight from ArcGIS json to RFC 7946 compliant GeoJSON, you'll also find the ESRI's javascript library for converting https://github.com/Esri/arcgis-to-geojson-utils/ conforms to the 2008 standard. This is why this library behaves the way it does - it mirrors the behaviour of the ESRI package I ported from :)

If you have an immediate need for a RFC 7946-compliant GeoJSON, you could always convert from ArcGIS using this library (or one of the other solutions mentioned) and then use GDAL to go from the 2008 spec to RFC 7946 as a post-process.

I don't have an pressing need for this myself, but I will have a think about what is involved and how to approach it.

**Comment by learning4life** [comment]
Thanks Chris, I appreciate your reply!

**Comment by chris48s** [comment]
I'm sure you're following along at home, but the way I've decided to approach this is to submit changes to the ESRI's library and then once they are approved, I'll port them here. In some ways it is a bit of a roundabout way to do it, but I am keen to keep the behaviour of this library close to the ESRI implementation so it makes sense to contribute the changes upstream first :)

**Comment by chris48s** [comment]
I'm going to close this because I've now published [version 1.1.0](https://pypi.python.org/pypi/arcgis2geojson/1.1.0) to PyPI which tracks the upstream changes:

* Use right-hand rule polygon winding order
* Log a warning if input SRID is not 4326
* Don't allow 'id' which is not number or string

This implements the hard requirements of RFC 7946 and you should find that outputs produced by this library now pass when run through a validator or linter which checks against the spec :)
